### PR TITLE
Refactor size queries with new enum SizeQueryType

### PR DESCRIPTION
**Pull request by agedemenli**
DESCRIPTION: Refactors size queries with new enum SizeQueryType

### CODE REVIEW

File: src/backend/distributed/metadata/metadata_utility.c [hunk 1]
```diff
@@ -614,7 +617,24 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList, char *sizeQuery)
...
+		}
+		else if (sizeQueryType == CSTORE_TABLE_SIZE)
+		{
+			appendStringInfo(selectQuery, CSTORE_TABLE_SIZE_FUNCTION, quotedShardName);
+		}
+		else if (sizeQueryType == TABLE_SIZE)
+		{
+			appendStringInfo(selectQuery, PG_TABLE_SIZE_FUNCTION, quotedShardName);
+		}
+
```
Comment (agedemenli) [inline src/backend/distributed/metadata/metadata_utility.c:637]: Consider a switch statement for here

File: src/backend/distributed/metadata/metadata_utility.c [hunk 2]
```diff
@@ -452,7 +453,8 @@ DistributedTableSize(Oid relationId, char *sizeQuery, bool failOnError, uint64 *
  * shard placement.
  */
 static bool
-DistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId, char *sizeQuery,
+DistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId, SizeQueryType
+							 sizeQueryType,
```
Comment (thanodnl) [inline src/backend/distributed/metadata/metadata_utility.c:457]: hit; Lets keep type and variable on the same line

```suggestion
DistributedTableSizeOnWorker(WorkerNode *workerNode, Oid relationId,
							 SizeQueryType sizeQueryType,
```

File: src/backend/distributed/metadata/metadata_utility.c (hunk 1 above)
Comment (thanodnl) [reply src/backend/distributed/metadata/metadata_utility.c:637]: I was about to suggest a switch. The default branch could use an `elog(ERROR,....)` as it should be expected the user will never hit it.

File: src/include/distributed/metadata_utility.h [hunk 3]
```diff
@@ -177,6 +177,16 @@ typedef struct TableConversionReturn
 }TableConversionReturn;
 
 
+/* Size queries for PG and Citus*/
+typedef enum SizeQueryType
+{
+	RELATION_SIZE,
+	TOTAL_RELATION_SIZE,
+	TABLE_SIZE,
```
Comment (thanodnl) [inline src/include/distributed/metadata_utility.h:185]: what is the difference between a `RELATION_SIZE` and a `TABLE_SIZE`.
I do see that the original code called `pg_table_size` is the result of that function different compared to `pg_relation_size` ?

A comment on these values describing their subtile differences would be beneficial.

**Review by thanodnl — CHANGES_REQUESTED** [review]
small changes required.

switch statement please,
some comments on the difference between tablesize and relation size. If they are the same, now would be a great moment to dedupe them into the relation size. However if there are differences documenting them is enough.

File: src/backend/distributed/metadata/metadata_utility.c [hunk 4]
```diff
@@ -628,6 +634,39 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList, char *sizeQuery)
...
+		case TABLE_SIZE:
+		{
+			return PG_TABLE_SIZE_FUNCTION;
+		}
+		default:
+		{
+			elog(ERROR, "Size query type couldn't be found.");
+		}
+	}
+}
```
Comment (thanodnl) [inline src/backend/distributed/metadata/metadata_utility.c:667]: Nice, was thinking about suggesting the extraction in a function, but didn't want to enforce it. You taking this initiative yourself is great!

**Review by thanodnl — APPROVED** [review]
Please fix the style violations that CI pointed out and it will be ready to merge once all tests pass.

**Pull request merged**